api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java

```diff
@@ -326,10 +326,128 @@ public CohortDefinition getPatientsWhoSuspendedTreatment() {
+            + "              AND e.location_id = :locationList "
+            + "              AND e.encounter_type = ${masterCard} ";
+    if (data1 == true) {
+      sql = sql + "AND o.obs_datetime BETWEEN :onOrAfter AND :onOrBefore ";
+    } else if (data1 == false) {
+      sql = sql + "AND o.obs_datetime <= :onOrBefore ";
+    }
+    sql =
+        sql
+            + "AND o.voided = 0 "
+            + "              AND o.concept_id = ${preArtStateOfStay}  "
+            + "              AND o.value_coded = ${patientDeadConcept} "
+            + "      GROUP BY patient_id"
+            + "            UNION"
+            + "       SELECT pg.patient_id, ps.start_date"
+            + "       FROM patient p"
+            + "         INNER JOIN patient_program pg ON p.patient_id=pg.patient_id"
+            + "         INNER JOIN patient_state ps ON pg.patient_program_id=ps.patient_program_id "
+            + "       WHERE pg.voided=0 AND ps.voided=0 AND p.voided=0 "
+            + "         AND pg.program_id=${arvProgram}  AND ps.state=${deadState} AND ps.end_date is null ";
+    if (data1 == true) {
+      sql = sql + "AND ps.start_date BETWEEN :onOrAfter AND :onOrBefore ";
+    } else if (data1 == false) {
+      sql = sql + "AND ps.start_date <= :onOrBefore ";
+    }
+    sql =
+        sql
+            + "AND location_id=:locationList"
+            + "            UNION"
+            + "        SELECT p.person_id patient_id, p.death_date "
+            + "      FROM person p "
+            + "      WHERE p.dead=1 ";
+    if (data1 == true) {
+      sql = sql + "AND p.death_date BETWEEN :onOrAfter AND :onOrBefore ";
+    } else if (data1 == false) {
+      sql = sql + "AND p.death_date <= :onOrBefore ";
+    }
+    sql =
+        sql
+            + "AND p.voided=0) dead"
+            + "       GROUP  BY patient_id) all_dead"
+            + "      WHERE patient_id NOT IN (SELECT p.patient_id "
+            + "         FROM patient p "
+            + "         JOIN encounter e "
+            + "         ON p.patient_id = e.patient_id "
+            + "       WHERE  p.voided = 0 "
+            + "       AND e.voided = 0 "
+            + "       AND e.encounter_type IN ( ${adultoSeguimento}, ${pediatriaSeguimento}, ${farmacia} )  "
+            + "       AND e.location_id = :locationList "
+            + "       AND e.encounter_datetime > death_date "
+            + "       UNION "
+            + "       SELECT p.patient_id "
+            + "       FROM   patient p "
+            + "       JOIN encounter e "
+            + "       ON p.patient_id = e.patient_id "
+            + "       JOIN obs o "
+            + "       ON e.encounter_id = o.encounter_id "
+            + "       WHERE  p.voided = 0 "
+            + "       AND e.voided = 0 "
+            + "       AND e.encounter_type = ${arvLevantamento} "
+            + "       AND e.location_id = :locationList "
+            + "       AND o.concept_id = 23866 "
```

Comment:
Concept id should not be hard-coded, please use a template string.

Reply:
Done